hazelcast/src/main/java/com/hazelcast/crdt/pncounter/PNCounter.java

```diff
@@ -0,0 +1,216 @@
+
+/**
+ * PN (Positive-Negative) CRDT counter.
+ * <p>
+ * The counter supports adding and subtracting values as well as
+ * retrieving the current counter value.
+ * Each replica of this counter can perform operations locally without
+ * coordination with the other replicas, thus increasing availability.
+ * The counter guarantees that whenever two nodes have received the
+ * same set of updates, possibly in a different order, their state is
+ * identical, and any conflicting updates are merged automatically.
+ * If no new updates are made to the shared state, all nodes that can
+ * communicate will eventually have the same data.
+ * <p>
+ * Each replica is identified by a UUID. This identifier should be
+ * unique and there should not be any other replica with the same ID, even
+ * if the member with that ID is no longer alive.
+ * <p>
+ * The updates to this counter are applied locally when invoked on a
+ * CRDT replica. A replica can be any hazelcast instance which is not a
+ * client or a lite member. The number of replicas in the cluster is
+ * determined by the {@link PNCounterConfig#getReplicaCount()} configuration
+ * value.
+ * <p>
+ * When invoking updates from non-replica instance, the invocation is remote.
+ * This may lead to indeterminate state - the update may be applied but the
+ * response has not been received. In this case, the caller will be notified
+ * with a {@link com.hazelcast.spi.exception.TargetDisconnectedException}.
+ * <p>
+ * The read and write methods provide monotonic read and RYW (read-your-write)
+ * guarantees. These guarantees are session guarantees which means that if
+ * no replica with the previously observed state is reachable, the session
+ * guarantees are lost and the method invocation will throw a
+ * {@link com.hazelcast.crdt.ConsistencyLostException}. This does not mean
+ * that an update is lost. All of the updates are part of some replica and
+ * will be eventually reflected in the state of all other replicas. This
+ * exception just means that you cannot observe your own writes because
+ * all replicas that contain your updates are currently unreachable.
+ * <p>
+ * <b>NOTE:</b>
+ * The CRDT state is kept entirely on non-lite (data) members. If there
+ * aren't any and the methods here are invoked on a lite member, they will
+ * fail with an {@link MutationDisallowedException}.
+ *
+ * @since 3.10
+ */
+public interface PNCounter extends DistributedObject {
+    /**
+     * Returns the current value of the counter.
+     *
+     * @throws MutationDisallowedException                 if the cluster does
+     *                                                     not contain any data
+     *                                                     members
+     * @throws UnsupportedOperationException               if the cluster version
+     *                                                     is less than 3.10
+     * @throws com.hazelcast.crdt.ConsistencyLostException if the session guarantees
```

Comment:
minor comment: we can import `ConsistencyLostException`

Reply:
Fixed.